include/sundials/sundials_matrix.h

```diff
@@ -91,6 +92,9 @@ struct _generic_SUNMatrix_Ops {
   int          (*matvecsetup)(SUNMatrix);
   int          (*matvec)(SUNMatrix, N_Vector, N_Vector);
   int          (*space)(SUNMatrix, long int*, long int*);
+#ifdef __cplusplus
+  _generic_SUNMatrix_Ops() = default;
+#endif
```

Comment:
Do the other classes need this as well (vector, linear solver, and nonlinear solver)?